avm/ptn/sa/conversation-knowledge-mining/main.bicep

```diff
@@ -0,0 +1,1168 @@
+      { name: 'AZURE-SEARCH-ENDPOINT', value: 'https://${varAiFoundrySearchServiceResourceName}.search.windows.net' }
+      { name: 'AZURE-SEARCH-SERVICE', value: varAiFoundrySearchServiceResourceName }
+      { name: 'AZURE-SEARCH-INDEX', value: 'transcripts_index' }
+      { name: 'COG-SERVICES-NAME', value: varAiFoundryAiServiceResourceName }
+      { name: 'AZURE-SUBSCRIPTION-ID', value: subscription().subscriptionId }
+      { name: 'AZURE-RESOURCE-GROUP', value: resourceGroup().name }
+      { name: 'AZURE-LOCATION', value: solutionLocation }
+    ]
+  }
+}
+
+// ========== AI Foundry ========== //
+
+module moduleAIFoundry './modules/ai-foundry.bicep' = {
+  name: format(localModuleDeploymentNameFormat, 'ai-foundry')
+  params: {
+    aiHubLocation: aiFoundryAiHubLocation
+    aiHubResourceName: varAiFoundryAiHubResourceName
+    aiHubSkuName: aiFoundryAiHubSkuName
+    aiProjectLocation: aiFoundryAiProjectLocation
+    aiProjectResourceName: varAiFoundryAiProjectResourceName
+    aiProjectSkuName: aiFoundryAiProjectSkuName
+    aiServicesLocation: aiFoundryAiServicesLocation
+    aiServicesContentUnderstandingLocation: aiFoundryAiServicesContentUnderstandingLocation
+    aiServicesContentUnderstandingResourceName: varAiFoundryAiServicesContentUnderstandingResourceName
+    aiServicesContentUnderstandingSkuName: aiFoundryAiServicesContentUnderstandingSkuName
+    aiServicesModelDeployments: varAiFoundryAIServicesModelDeployments
+    aiServicesResourceName: varAiFoundryAiServiceResourceName
+    aiServicesSkuName: aiFoundryAiServicesSkuName
+    applicationInsightsLocation: aiFoundryApplicationInsightsLocation
+    applicationInsightsResourceName: varAiFoundryApplicationInsightsResourceName
+    applicationInsightsRetentionInDays: aiFoundryApplicationInsightsRetentionInDays
+    containerRegistryLocation: aiFoundryContainerRegistryLocation
+    containerRegistryResourceName: varAiFoundryContainerRegistryResourceName
+    containerRegistrySkuName: aiFoundryContainerRegistrySkuName
+    enableTelemetry: enableTelemetry
+    keyVaultResourceName: avmKeyVault.outputs.name
+    logAnalyticsWorkspaceResourceId: avmLogAnalyticsWorkspace.outputs.resourceId
+    managedIdentityPrincipalId: avmManagedIdentity.outputs.principalId
+    searchServiceLocation: aiFoundrySearchServiceLocation
+    searchServiceResourceName: varAiFoundrySearchServiceResourceName
+    searchServiceSkuName: aiFoundrySearchServiceSkuName
+    storageAccountLocation: aiFoundryStorageAccountLocation
+    storageAccountResourceName: varAiFoundryStorageAccountResourceName
+    storageAccountSkuName: aiFoundryStorageAccountSkuName
+    tags: tags
+    avmDeploymentNameFormat: avmDeploymentNameFormat
+    localModuleDeploymentNameFormat: localModuleDeploymentNameFormat
+  }
+}
+
+// ========== Storage Account ========== //
+
+module avmStorageAccount 'br/public:avm/res/storage/storage-account:0.18.2' = {
```

Comment:
I'd recommend to double-check but I think you can remove a large chunk of the parmaeters below as they're defaults of the module and or child-modules. The same may be true for other AVM module deployments.

Reply:
This may not have been the case for all. You don't **have to** change the implementation, but if you want you could, for example, remove:
- supportsHttpsTrafficOnly
- enableNfsV3
- automaticSnapshotPolicyEnabled
- deleteRetentionPolicyAllowPermanentDelete
- ...

Reply:
The same applies to other template files, btw.

And before you ask: Yes, I already requested to please add parameter defaults to the Bicep Intellisense support - but it is yet to be implemented (i.e., you only know of the defaults by checking the value in the targeted template)

Reply:
Gone a second round and removed a few redundant default configurations that I missed in my first iteration :)